src/xref2/lookup_failures.mli

```diff
@@ -13,6 +13,11 @@ val report_important :
 (** Like [report] above but may raise the exception [exn] if strict mode is
     enabled *)
 
+val report_warning : Odoc_model.Error.t -> unit
+(** Report a warning to the enclosing [catch_failures].
+    The warning will be displayed before other failures. *)
```

Comment:
This changes `Lookup_failures` to also accumulate warnings in order to avoid using the 2 mechanisms at the same time. It may not be a good idea. What do you think ?

The differences between `Lookup_failures` and the new mechanism in `Odoc_model.Error` are:
- Lookup failures are aggregated into a single message, which is then printed as a warning
- Lookup failures could be fatal and abort compilation in "strict mode", which is not implemented yet